File: Algorithms/C++/String/14_Longest_Common_Prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string prefix;

        if (strs.empty()) {
            return prefix;
        }

        int min_len = INT_MAX;
        for (int i = 0; i < strs.size(); i++) {
            if (strs[i].length() < min_len) {
                min_len = strs[i].length();
            }
        }

        for (int i = 0; i < min_len; i++) {
            for (int j = 1; j < strs.size(); j++) {
                if (strs[j][i] != strs[0][i]) {
                    return prefix;
                }
            }

            prefix += strs[0][i];
        }

        return prefix;
    }
};

File: Algorithms/Swift/String/14_Longest_Common_Prefix.swift
class Solution {
    func longestCommonPrefix(strs: [String]) -> String {
        if strs.isEmpty {
            return ""
        }

        var commonPrefix = ""

        var minLength = Int.max
        for str in strs {
            if str.characters.count < minLength {
                minLength = str.characters.count
            }
        }

        for var i = 0; i < minLength; i++ {
            let ch = strs[0][i]

            var j = 1;
            for ; j < strs.count && strs[j][i] == ch; j++ {

            }

            if j == strs.count {
                commonPrefix.append(ch)
            } else {
                return commonPrefix
            }
        }

        return commonPrefix;
    }
}

extension String {
    subscript(i: Int) -> Character {
        return self[self.startIndex.advancedBy(i)]
    }
}
